tensorbay/opendataset/CityscapesGTFine/loader.py

```diff
@@ -0,0 +1,110 @@
+
+"""Dataloader of CityscapesGTFine dataset."""
+import json
+import os
+from glob import glob
+from typing import List
+
+from tensorbay.dataset import Data, Dataset
+from tensorbay.label import InstanceMask, LabeledPolygon, SemanticMask
+
+DATASET_NAME = "CityscapesGTFine"
+_SEGMENTS_INFO = {"test", "train", "val"}
+
+
+def CityscapesGTFine(path: str) -> Dataset:
+    """`CityscapesGTFine <https://www.cityscapes-dataset.com/>`_ dataset.
+
+    The file structure should be like::
+
+        <path>
+            leftImg8bit/
+                test/
+                    berlin/
+                        berlin_000000_000019_leftImg8bit.png
+                        ...
+                    ...
+                train/
+                    aachen/
+                        aachen_000000_000019_leftImg8bit.png
+                        ...
+                    ...
+                val/
+                    frankfurt/
+                        frankfurt_000000_000019_leftImg8bit.png
+                        ...
+                    ...
+                ...
+            gtFine/
+                test/
+                    berlin/
+                        berlin_000000_000019_gtFine_instanceIds.png
+                        berlin_000000_000019_gtFine_labelIds.png
+                        berlin_000000_000019_gtFine_polygons.json
+                        ...
+                    ...
+                train/
+                    aachen/
+                        aachen_000000_000019_gtFine_instanceIds.png
+                        aachen_000000_000019_gtFine_labelIds.png
+                        aachen_000000_000019_gtFine_polygons.json
+                        ...
+                    ...
+                val/
+                    frankfurt/
+                        frankfurt_000000_000019_gtFine_instanceIds.png
+                        frankfurt_000000_000019_gtFine_labelIds.png
+                        frankfurt_000000_000019_gtFine_polygons.json
+                        ...
+                    ...
+            ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.join(os.path.abspath(os.path.expanduser(path)))
+
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for segment_name in _SEGMENTS_INFO:
+        segment = dataset.create_segment(segment_name)
+        image_paths = glob(os.path.join(root_path, "leftImg8bit", "*", "*", "*.png"))
+        for image_path in image_paths:
+            city = image_path.split("_")[0]
+            image_prefix = image_path.rsplit("_", 1)[0]
+            data = Data(image_path)
+            # get semantic mask and instance mask
+            label = data.label
+            label.semantic_mask = SemanticMask(
+                os.path.join(root_path, "gtFine", segment_name, city, image_prefix, "_labelIds.png")
```

Comment:
it seems that concatenating `image_prefix` with `"_labelIds.png"` will give you the mask filename, so it should be a f-string instead of two args of `join()`